### DISCUSSION TITLE
Command not showing in experimental instance

### DISCUSSION THREAD

**Discussion by asherber** [VsixCommunity/Community.VisualStudio.Toolkit#214] [Q&A]
I'm just starting to play around with extensions, but I can't get off the ground.

I've got VS2022 Community installed with extensibility workload. I went to the cookbook site and installed Extensibility Essentials 2022. I started a new solution and added a "VSIX Project w/Command (Community)" project. The cookbook has info about modifying the project, but as I understand it, even without modifying I should be able to F5 on the new project and see an entry in the Tools menu of the experimental instance that says "My Command", and when I click that item I should get a dialog that says "Button clicked". But the menu entry is not showing up for me.

What am I missing?

**Comment by moon6969** [comment]
I've just done several such tests today on 2 PCs - for me the "VSIX Project w/Command (Community)" project does just F5 out of the box.

Does the extension appear in the _Extensions → Manage Extensions_ dialog (in the Experimental Instance when debugging)?

Have you tried "Reset the Visual Studio 2022 Experimental Instance" (in the Start Menu)?

Since you posted there have been updates to the 'Extensibility Essentials' and VSSDK build NuGet packages, so it may be worth updating them in your test solution...
https://www.vsixcookbook.com/getting-started/get-the-tools.html

**Reply by asherber** [reply]
@moon6969 Thanks for following up, but this still isn't working for me. My VS is up to date, I reset the experimental instance, and I started a new solution with "VSIX Project w/Command (Community)". When I hit F5, the experimental instance opens, but the new extension does not appear in the Manage Extensions dialog, and the menu item does not appear under Tools.

I'm an experienced dev but new to VS extensions -- not sure what I could be doing wrong.

**Reply by moon6969** [reply]
> My VS is up to date

'Extensibility Essentials' extension needs to be updated seperately from VS via _Extensions → Manage Extensions → Updates_. 
Mine currently...
- 'Extensibility Essentials 2022'  v1.1.14.

Similary check _Tools → Nuget Package Manager → Manage Nuget packages for Solution_ to see if there are updates to the community or SDK packages.
Mine currently...
- Community.VisualStudio.VSCT       {16.0.29.6} 
- Community.VisualStudio.Toolkit.17 {17.0.394}  
- Microsoft.VSSDK.BuildTools        {17.1.4054}

**Reply by asherber** [reply]
All of those are up to date (plus all the extensions that Extensibility Essentials is an umbrella for). Still no luck.

**Reply by moon6969** [reply]
A few other random thoughts...

- Is the .vsix created in the bin folder?
- Is there anything interesting in the MSBuild log (increase the verbosity _Tools → Options → Projects and Solutions → Build and Run → MSBuild project build output verbosity: Diagnostic_)
- Do the extension binaries, manifests etc. appear under: 
`%LOCALAPPDATA%\Microsoft\VisualStudio\17.0_8548fb75Exp\Extensions\<<Author>>\VSIXProject1\1.0`
- Can you manually install the .vsix file by double-clicking it (⚠️this will install to your live VS not the experimental instance - test it on a virtual box if this is of concern).
- Is it possible another extension is causing issues? If you're feeling brave (or are testing on a virtual box), you could try deleting the extensions under
` C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\Extensions`

**Reply by asherber** [reply]
- .vsix is in bin folder
- Nothing of note in the build log
- Extension files appear in `%localappdata%...` folder
- Am able to manually install .vsix into live VS -- it shows up in the Tools menu and works
- Disabled all current extensions, reset experimental instance, and tried again -- still no luck

**Reply by moon6969** [reply]
It must be frustrating. At this point I would typically go 🤯 and do a fresh install of Windows.
Hope you sort it out.

**Reply by ErikEJ** [reply]
Do you target AnyCPU?

**Reply by asherber** [reply]
@ErikEJ Yes, AnyCPU

@moon6969 I'm not ready to pave my machine on account of this. 

I spun up a new AWS EC2 and installed VS, and the VSIX Community project does work out of the box with F5 (as I would expect).

However, I'm having this issue on both of my machines -- a Dell desktop and an HP laptop, both up to date with Windows and VS. I do keep them similarly configured, so at this point I'm thinking it's either some VS setting that I have or something else that I have running on both machines. Is there some way to have the experimental instance launch with the `/log` switch?

**Reply by StevenRasmussen** [reply]
@asherber - I never saw if you confirmed that you can see the extension in the `Extensions -> Manage Extensions` dialog?  If so, is it enabled?
Also, you *should* be able to run the experimential instance by putting the `/log` switch at the end of the `Startup external program:` in the settings of the VSIX project:

![image](https://user-images.githubusercontent.com/11198735/159130730-89f08c06-3478-4024-8fdb-b0ba4cc0f881.png)

**Reply by asherber** [reply]
@StevenRasmussen The extension does not show up in `Extensions -> Manage Extensions`

VS doesn't like it if you add command switches in that spot. I tried adding it in the `Command line arguments` edit box, but VS deleted it as soon as I hit F5.

**Reply by StevenRasmussen** [reply]
Have you checked the following in the properties for the project:
![image](https://user-images.githubusercontent.com/11198735/159184968-4ebe2808-13f6-4a78-8304-d3860379ebe8.png)

**Reply by asherber** [reply]
@StevenRasmussen Yup. All I'm basically doing right now is starting a new "VSIX Project w/Command (Community)" and hitting F5, expecting the "My Command" item to show up in the Tools menu. It works for me on a clean EC2, and presumably is working fine for lots of other people, but it's not working on either of my two (similarly configured) computers.
